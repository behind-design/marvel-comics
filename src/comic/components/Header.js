import React from 'react';


const Header = ({ comic }) => (
  <header>
    <div className="top">
      <h1>{comic.title} #{comic.issueNumber}</h1>
      <p>{comic.year}</p>
    </div>

    <p>{comic.price}</p>
  </header>
);

export default Header;
