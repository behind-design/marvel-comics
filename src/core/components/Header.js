import React from 'react';
import { Link } from 'react-router-dom';
import cn from 'classnames';


const Header = ({ detail }) => (
  <div className={cn({
    'marvel-header': true,
    detail,
  })}>
    <Link to="/">
      <div id="marvel-logo">
        <img src="/img/marvel-logo.svg" alt="Marvel Comics"/>
      </div>
    </Link>
  </div>
)

export default Header;
