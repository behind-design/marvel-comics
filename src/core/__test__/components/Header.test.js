import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { mount, render } from 'enzyme';

import Header from '../../components/Header';


describe('Header', () => {
  it('renders without crashing', () => {
    mount(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
  });

  it('renders all the correct properties', () => {
    const wrapper = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with the correct className', () => {
    const wrapper = render(
      <MemoryRouter>
        <Header detail={true} />
      </MemoryRouter>
    )

    expect(wrapper).toMatchSnapshot();
  });
});
