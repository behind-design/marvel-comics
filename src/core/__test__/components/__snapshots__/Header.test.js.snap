// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders all the correct properties 1`] = `
<div
  class="marvel-header"
>
  <a
    href="/"
  >
    <div
      id="marvel-logo"
    >
      <img
        alt="Marvel Comics"
        src="/img/marvel-logo.svg"
      >
        
      </img>
    </div>
  </a>
</div>
`;

exports[`Header renders with the correct className 1`] = `
<div
  class="marvel-header detail"
>
  <a
    href="/"
  >
    <div
      id="marvel-logo"
    >
      <img
        alt="Marvel Comics"
        src="/img/marvel-logo.svg"
      >
        
      </img>
    </div>
  </a>
</div>
`;
